drivers/mipi_dsi/dsi_mcux.c

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2022, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_imx_mipi_dsi
+
+#include <drivers/video.h>
+#include <drivers/mipi_dsi.h>
+#include <fsl_mipi_dsi.h>
+#include <fsl_clock.h>
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(dsi_mcux, CONFIG_MIPI_DSI_LOG_LEVEL);
+
+/*
+ * The DPHY bit clock must be fast enough to send out the pixels, it should be
+ * larger than:
+ *
+ *         (Pixel clock * bit per output pixel) / number of MIPI data lane
+ *
+ * Here the desired DPHY bit clock multiplied by ( 9 / 8 = 1.125) to ensure
+ * it is fast enough.
+ */
+#define MIPI_DPHY_BIT_CLK_ENLARGE(origin) (((origin) / 8) * 9)
+
+struct display_mcux_mipi_dsi_config {
+	MIPI_DSI_Type base;
+	dsi_dpi_config_t dpi_config;
+	bool auto_insert_eotp;
+};
+
+struct display_mcux_mipi_dsi_data {
+	const struct device *dev;
+};
+
+static int dsi_mcux_attach(const struct device *dev,
+			   uint8_t channel,
+			   const struct mipi_dsi_device *mdev)
+{
+	const struct display_mcux_mipi_dsi_config *config = dev->config;
+	dsi_dphy_config_t dphyConfig;
+	dsi_config_t dsiConfig;
+	uint32_t mipiDsiEscClkFreq_Hz;
+	uint32_t mipiDsiTxEscClkFreq_Hz;
+	uint32_t mipiDsiDphyRefClkFreq_Hz = 24000000;
```

Comment:
magic number here, why 24MHz?
another comment: please don't use `camelCase` naming in Zephyr code.

Reply:
Thanks for catching. I moved this to device tree. Also I have fixed the `camelCase` naming of variables